### ISSUE TITLE
postserialize hook doesn't respect image serializers

### ISSUE THREAD

**Issue by blairj09** [rstudio/plumber#517]
### System details

Output of `sessioninfo::session_info()`:

``` r
sessioninfo::session_info()
─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       macOS Catalina 10.15.3      
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Denver              
 date     2020-03-27                  

─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────────
 ! package     * version     date       lib source                           
 P assertthat    0.2.1       2019-03-21 [?] CRAN (R 3.6.0)                   
 P cli           2.0.2       2020-02-28 [?] CRAN (R 3.6.2)                   
 P crayon        1.3.4       2017-09-16 [?] CRAN (R 3.6.0)                   
 P fansi         0.4.1       2020-01-08 [?] CRAN (R 3.6.0)                   
 P fs            1.3.2       2020-03-05 [?] CRAN (R 3.6.0)                   
 P glue        * 1.3.2       2020-03-12 [?] CRAN (R 3.6.3)                   
   httpuv        1.5.2.9000  2020-03-27 [1] Github (rstudio/httpuv@622c76a)  
 P jsonlite      1.6.1       2020-02-02 [?] CRAN (R 3.6.0)                   
 P later         1.0.0       2019-10-04 [?] CRAN (R 3.6.0)                   
 P logger      * 0.1         2019-01-02 [?] CRAN (R 3.6.0)                   
 P magrittr      1.5         2014-11-22 [?] CRAN (R 3.6.0)                   
   plumber     * 0.4.7.9000  2020-03-27 [1] Github (rstudio/plumber@54fc7f7) 
   promises      1.1.0.9000  2020-03-27 [1] Github (rstudio/promises@627dfc6)
 P R6            2.4.1       2019-11-12 [?] CRAN (R 3.6.0)                   
   Rcpp          1.0.4.5     2020-03-27 [1] Github (RcppCore/Rcpp@7416859)   
   renv          0.9.3       2020-02-10 [1] CRAN (R 3.6.2)                   
 P rstudioapi    0.11        2020-02-07 [?] CRAN (R 3.6.0)                   
 P sessioninfo   1.1.1       2018-11-05 [?] CRAN (R 3.6.0)                   
   stringi       1.4.6       2020-02-17 [1] CRAN (R 3.6.0)                   
   swagger       3.20.3.9999 2020-03-27 [1] Github (rstudio/swagger@67239fd) 
 P withr         2.1.2       2018-03-15 [?] CRAN (R 3.6.0)                   

[1] /Users/jamesblair/Documents/RStudio/repos/plumber-examples/renv/library/R-3.6/x86_64-apple-darwin15.6.0
[2] /private/var/folders/d_/_k539mwd3xsg0chwvt782vzm0000gn/T/RtmpvitORD/renv-system-library

 P ── Loaded and on-disk path mismatch.
```

### Example application

`plumber.R`
```r
library(plumber)

#* @apiTitle Plumber Image Serializer

#* Echo back the input
#* @param msg The message to echo
#* @get /echo
function(msg = "") {
    list(msg = paste0("The message is: '", msg, "'"))
}

#* Plot a histogram
#* @png
#* @get /plot-png
function() {
  rand <- rnorm(100)
  hist(rand)
}

#* Plot a histogram
#* @jpeg
#* @get /plot-jpeg
function() {
  rand <- rnorm(100)
  hist(rand)
}
```

`entrypoint.R`
```r
library(plumber)

pr <- plumb("plumber.R")

pr$registerHooks(
  list(
    postserialize = function(req, res) {
      print("Postserialize Hook")
    }
  )
)

pr
```

### Describe the problem in detail
When either `/plot` endpoint is invoked, the postserialize hook doesn't execute, as evidenced by the fact that nothing is printed to the console. When the `/echo` endpoint is invoked, `Postserialize Hook` appears in the console, as expected.

**Comment by meztez** [comment]
I would try a browser() to check if the hook does not register. I could not reproduce the behavior.

```
library(plumber)

pr <- plumb()

pr$registerHooks(
  list(
    postserialize = function(req, res) {
      cat("Postserialize Hook cat")
      print("Postserialize Hook print")
      browser()
    }
  )
)

pr$run()
```

**Comment by meztez** [comment]
It is not limited to plot. Behavior is different from the CRAN version. It does not get to the postserialer functions since it returns from https://github.com/rstudio/plumber/blob/23f81b8d2229e4965ac239138a5bceb61894e03d/R/plumber.R#L441

**Comment by meztez** [comment]
As I understand it, there is no actual serialization happening when png or jpeg are used. The content of the graphical output is captured in a file and that file becomes the body of the response.

Once it reach [441](https://github.com/rstudio/plumber/blob/23f81b8d2229e4965ac239138a5bceb61894e03d/R/plumber.R#L441) it exits since the job is already done and the pre/post serialization are not executed.

You could a postroute step but I agree this is a regression from previous behavior

What I don't understand yet is why the CRAN version would still executes the postserialization steps.

**Comment by schloerke** [comment]
This is fixed in v1.0.0. Closing
